### PR TITLE
(SIMP-2762) Fully Parameterized README for svckill

### PR DESCRIPTION
**Pull request by RARYates**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,260 @@
...
+
+1. [Overview](#this-is-a-simp-module)
+2. [Module Description - A Puppet module for managing svckill](#module-description)
+3. [Setup - The basics of getting started with pupmod-simp-svckill](#setup)
+    * [What pupmod-simp-svckill affects](#what-simp-svckill-affects)
+    * [Setup requirements](#setup-requirements)
+    * [Beginning with svckill](#beginning-with-svckill)
+4. [Usage - Configuration options and additional functionality](#usage)
+5. [Reference - An under-the-hood peek at what the module is doing and how](#reference)
+5. [Limitations - OS compatibility, etc.](#limitations)
```
Comment (op-ct) [inline README.md]: x2 :hash::five:

File: README.md [hunk 2]
```diff
@@ -1,14 +1,260 @@
...
+   These are here because their status function is broken
+      * krb524
+      * mdmpd
+      * readahead_later
+      * rawdevices
+      * lm_sensors
+      * kudzu
+```
+
+### Setup Requirements
```
Comment (op-ct) [inline README.md]: **Question:** should these "`###`" headings be visible from the TOC like the ones from the _Setup_ section are?

File: README.md [hunk 3]
```diff
@@ -1,14 +1,260 @@
...
 
-If you find any issues, they can be submitted to our [JIRA](https://simp-project.atlassian.net/).
+## Limitations
 
-Please read our [Contribution Guide](https://simp-project.atlassian.net/wiki/display/SD/Contributing+to+SIMP) and visit our [developer wiki](https://simp-project.atlassian.net/wiki/display/SD/SIMP+Development+Home).
+This module is only designed to work in RHEL or CentOS 6 and 7. Any other
+operating systems have not been tested and results cannot be guaranteed.
 
-## Work in Progress
+# Development
```
Comment (op-ct) [inline README.md]: To fit in with the other top-level TOC items, this should be a "`##`" level heading.

File: README.md [hunk 4]
```diff
@@ -1,14 +1,260 @@
...
+
+### I want to ignore a list of services I deploy in a file
+
+```puppet
+svckill { 'ourservices':
+  ignorefiles => '/opt/services',
+}
+```
+
+### I want to set a list of services allowed in my entire infrascture
```
Comment (jeefberkey) [inline README.md]: I think you should have a section setting svckill::ignore in hiera. It might be easier for some people.

Also, infrastructure.

**Comment by RARYates** [comment]
So, we worked through it and talked about how we wanted to structure it.

Users who are used to Hiera are used to looking through the forge and
seeing "regular puppet code" and translating it to Hiera. Those who haven't
implemented Hiera yet, or have done it in a way that doesn't involve YAML
(Such as databasing or automatically generated JSON) are less familiar with
what they're looking at.

We decided for clarity, we'd stick with ol' fashioned puppet code.

**Comment by jeefberkey** [comment]
Chris's comments are still valid

File: README.md (hunk 2 above)
Comment (RARYates) [inline README.md]: Following other examples out there, it's really only the setup section that shows the third level TOC. Usage can change, and linking them in the TOC isn't common. I'm not opposed to it though, if it's something everyone else wants to see.

**Pull request merged**